apply-roles

```diff
@@ -342,24 +446,46 @@ main() {
   # retrieve Service principal id
   service_principal_id=$(az ad sp show --id "${application_id}" --query "id" --output tsv)
 
-  if [[ -n "${subscription_id}" ]]; then
+  readonly APPLY_ROLE
```

Comment:
⛏️ ♻️ IMO it is "cleaner" if you make the variable `readonly` as soon as possible after it has had it's value set, i.e. i suggest moving this to line 393.

Reply:
I see - I didn't know that. Updated!

Reply:
It is just so a developer that is reading the code doesn't have to scan so much code to determine if the value of the variable changed between when it is first set and the `readonly`. If there is less code to scan there is less cognitive load on the developer.